### PR TITLE
Add resolve command in favour of verify --no-verify

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Changes

Add resolve command in favour of verify --no-verify

### Testing

None, but upcoming @davidcok ?

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyDriver/Commands/VerifyCommand.cs [hunk 1]
```diff
@@ -22,3 +22,19 @@ class VerifyCommand : ICommandSpec {
     dafnyOptions.Compile = false;
   }
 }
+
+class ResolveCommand : ICommandSpec {
+  public IEnumerable<Option> Options => ICommandSpec.ConsoleOutputOptions.
+    Concat(ICommandSpec.CommonOptions);
+
+  public Command Create() {
+    var result = new Command("resolve", "Resolve the program.");
```
Comment (davidcok) [inline Source/DafnyDriver/Commands/VerifyCommand.cs:31]:
```suggestion
    var result = new Command("resolve", "Only check for parse and resolution errors.");
```

File: Source/DafnyDriver/Commands/TranslateCommand.cs [hunk 2]
```diff
@@ -11,8 +11,7 @@ class TranslateCommand : ICommandSpec {
       CommonOptionBag.Output,
       CommonOptionBag.CompileVerbose,
       CommonOptionBag.IncludeRuntime,
-    }.Concat(ICommandSpec.VerificationOptions).
```
Comment (davidcok) [inline Source/DafnyDriver/Commands/TranslateCommand.cs:14]: I would think that the translate command would still allow --no-verify

**Review by davidcok — CHANGES_REQUESTED** [review]
Thanks for this - I did agree with Mikael's comment.

File: Source/DafnyDriver/Commands/BuildCommand.cs [hunk 3]
```diff
@@ -9,8 +9,7 @@ class BuildCommand : ICommandSpec {
   public IEnumerable<Option> Options => new Option[] {
     CommonOptionBag.Output,
     CommonOptionBag.CompileVerbose,
-  }.Concat(ICommandSpec.VerificationOptions).
-    Concat(ICommandSpec.ExecutionOptions).
+  }.Concat(ICommandSpec.ExecutionOptions).
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/DafnyDriver/Commands/BuildCommand.cs:12]: Maybe we could tweak the formatter to enforce your preference? I'm ambivalent about both options.

**Review by davidcok — APPROVED** [review]

File: Source/DafnyDriver/Commands/VerifyCommand.cs (hunk 1 above)
Comment (MikaelMayer) [reply Source/DafnyDriver/Commands/VerifyCommand.cs]: Great suggestion !

**Review by MikaelMayer — APPROVED** [review]
Great job at separating the two concepts. Now my only (ultra-small) worry is that if a user makes dafny --help they might be presented the option to "resolve" first instead of "verify". But I'm not a user who uses the command line anyway so I don't have a strong opinion about what to do.
Let's keep this, this is an improvement already.

**Pull request merged**